frozen.js

```diff
@@ -0,0 +1,35 @@
+const cleanroom = () => Object.freeze(Object.create(null))
+const id = x => x
+
+const handlerFor = (store, { getter }) => ({
+  get: (_target, property, _receiver) => {
+    if (property === Symbol.iterator) {
+      return store[Symbol.iterator].bind(store)
+    }
+
+    if (store.has(property)) {
+      return getter(property)
+    }
+
+    throw new Error(`"${String(property)}" not found in this enum`)
+  },
+  set: () => { throw new Error("enums cannot be mutated") },
+})
+
+const FrozenSet = (ary) => {
+  const store = new Set(ary)
+
+  return new Proxy(cleanroom(), handlerFor(store, { getter: id }))
```

Comment:
I wrote a giant explanation and then accidentally closed the tab, so I have a pithy one for you.

We want to use an immutable object that has no properties or prototype chain of its own. `null` and `undefined` satisfy the lack of properties and prototype chain, but alas, they are not objects:

```es6
> new Proxy(null, {})
Uncaught TypeError: Cannot create proxy with a non-object as target or handler
> new Proxy(undefined, {})
Uncaught TypeError: Cannot create proxy with a non-object as target or handler
```

Instead, we must resort to creating an empty object with no prototype (`Object.create(null)`) and freeze it to prevent new property addition.

Reply:
This serves a similar function to Ruby's [`BasicObject`](https://ruby-doc.com/core/BasicObject.html)